### PR TITLE
initial working version

### PR DESCRIPTION
**Pull request by nelsonic**
## Ready for Review! :+1:

This Node Module Implements the feature described in: https://github.com/FAC-GM/app/issues/68

### CODE REVIEW

**Comment by nelsonic** [comment]
Done. See list of features in Readme. Please review and merge. Thanks. :+1:

File: lib/keyword-highlighter.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+/**
+ * Want to Set the Desired Start Tag to apply to the highlight?
+ * see: https://github.com/dwyl/search-result-keyword-highlighter/issues/5
+ * Potential future feature ...
+ */
+
+
+/**
+ * highilight does simple sting Substitution
```
Comment (SimonLab) [inline lib/keyword-highlighter.js]: Small typo, should be "string" instead of "sting"

File: lib/keyword-highlighter.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    } // can you think of a better way to find all the matches? Submit a PR! :-)
+
+    for(var i = matches.length-1; i > -1; i--) { // Good Old Zero-index ...
+      var sub = options.startTag + matches[i] + options.endTag;
+      var start = text.slice(0,matchpos[i]); // slice of text until keyword
+      var end   = text.slice(matchpos[i] + matches[i].length, text.length);
+      text = start + sub + end; // replace the text
+    }
+  }
+  console.log(text);
```
Comment (SimonLab) [inline lib/keyword-highlighter.js]: Do you still need a console.log here?

File: lib/keyword-highlighter.js (hunk 1 above)
Comment (nelsonic) [inline lib/keyword-highlighter.js]: Well spotted.

File: lib/keyword-highlighter.js (hunk 2 above)
Comment (nelsonic) [inline lib/keyword-highlighter.js]: another good find. glad someone is paying attention. :wink: 
Fixing now. :+1:

**Comment by nelsonic** [comment]
Hi @SimonLab, @rjmk has kindly re-factored the module to be _much_ cleaner. 
Yay TDD and having a Wizard to review our late-night code... :wink: 

There's an implicit function argument in the `replace()` when `wrapper` is called: https://github.com/dwyl/search-result-keyword-highlighter/blob/alpha/lib/keyword-highlighter.js#L31 which could use _clarifying comment_ 
for _complete beginners_. but other than that, its _exactly_ what we need. _optimised_. :sparkles:

Please review the change and merge if/when happy. thanks!

**Comment by SimonLab** [comment]
Looks good and neat! :ok_hand:

**Pull request merged**